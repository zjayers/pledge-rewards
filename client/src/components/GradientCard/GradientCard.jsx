/* Import Modules */
import React from 'react';
/* Import Components */
import Container from '../Shared/Container/Container';
import Heading from '../Shared/Heading/Heading';
import PledgeForm from '../PledgeForm/PledgeForm';
import GradientBox from './GradientCard.style';

/**
 * GradientCard Component
 * @returns {*}
 * @constructor
 */
const GradientCard = () => {
  /* Return the JSX to render */
  return (
    <Container activated>
      <Heading>Pledge without a reward</Heading>
      <PledgeForm activated>
        <GradientBox>
          <span>Back it because you believe in it.</span>
          <p>
            Support the project for no rewards, just because it speaks to you.
          </p>
        </GradientBox>
      </PledgeForm>
    </Container>
  );
};

/* Export the component - use the withTheme hook to allow theme access inside of this component */
export default GradientCard;
