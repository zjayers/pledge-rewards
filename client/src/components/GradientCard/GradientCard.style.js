import styled from 'styled-components';

export default styled.div`
  margin-top: 10px;
  padding: 1.8rem;
  width: 100%;
  height: auto;
  background: linear-gradient(to bottom, #dbe7ff, #fff2ec);
  border: 1px solid #dbe7ff;
  display: flex;
  flex-direction: column;
  align-items: center;

  & span,
  p {
    margin-bottom: 1.2rem;
    font-size: 1.4rem;
    line-height: 2.4rem;
  }

  & span {
    font-weight: 800;
    color: #222222;
  }
`;
