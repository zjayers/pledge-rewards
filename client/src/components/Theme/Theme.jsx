import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
