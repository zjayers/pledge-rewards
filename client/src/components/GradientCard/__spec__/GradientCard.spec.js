import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import GradientCard from '../GradientCard';
import Theme from '../../Theme/Theme';

describe('GradientCard Component', () => {
  let component;

  beforeEach(() => {
    component = mount(
      <Theme>
        <GradientCard />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
