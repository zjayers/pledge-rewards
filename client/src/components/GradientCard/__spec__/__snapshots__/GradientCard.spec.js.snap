// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientCard Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <GradientCard>
      <_default
        activated={true}
      >
        <Containerstyle___default
          activated={true}
        >
          <div
            className="Containerstyle___default-sc-1q8wjhx-0 BmWDh"
          >
            <_default>
              <Headingstyle___default>
                <h2
                  className="Headingstyle___default-dlpxpq-0 iErRtB"
                >
                  Pledge without a reward
                </h2>
              </Headingstyle___default>
            </_default>
            <WithTheme(Component)
              activated={true}
            >
              <Component
                activated={true}
                theme={
                  Object {
                    "colors": Object {
                      "borderGrey": "#9B9E9E",
                      "buttonGreen": "#009E74",
                      "flavorText": "#888B8B",
                      "gradientBlue": "#DBE7FE",
                      "gradientOrange": "#FFF2EC",
                      "highlightGreen": "#009E74",
                      "linkText": "#498F80",
                      "text": "#282828",
                    },
                    "fontSizes": Object {
                      "large": "3rem",
                      "medium": "1.8rem",
                      "small": "1rem",
                    },
                    "fonts": Array [
                      "Muli",
                      "sans-serif",
                    ],
                  }
                }
              >
                <_default
                  style={
                    Object {
                      "color": "#282828",
                    }
                  }
                  uppercase={false}
                >
                  <SubHeadingstyle___default
                    uppercase={false}
                  >
                    <p
                      className="SubHeadingstyle___default-qgikex-0 eiBTbF"
                    >
                      Pledge amount
                    </p>
                  </SubHeadingstyle___default>
                </_default>
                <CurrencyEntry
                  input=""
                  setInput={[Function]}
                >
                  <CurrencyEntrystyle__Form>
                    <div
                      className="CurrencyEntrystyle__Form-zl3z4f-0 cCvtGG"
                    >
                      <CurrencyEntrystyle__Input
                        min="0"
                        onChange={[Function]}
                        type="number"
                        value=""
                      >
                        <input
                          className="CurrencyEntrystyle__Input-zl3z4f-3 kOuXyP"
                          min="0"
                          onChange={[Function]}
                          type="number"
                          value=""
                        />
                      </CurrencyEntrystyle__Input>
                      <CurrencyEntrystyle__CurrencyBox>
                        <div
                          className="CurrencyEntrystyle__CurrencyBox-zl3z4f-2 jTmnGB"
                        >
                          <CurrencyEntrystyle__CurrencyIcon>
                            <span
                              className="CurrencyEntrystyle__CurrencyIcon-zl3z4f-1 evsVHR"
                            >
                              $
                            </span>
                          </CurrencyEntrystyle__CurrencyIcon>
                        </div>
                      </CurrencyEntrystyle__CurrencyBox>
                    </div>
                  </CurrencyEntrystyle__Form>
                </CurrencyEntry>
                <_default
                  uppercase={true}
                >
                  <SubHeadingstyle___default
                    uppercase={true}
                  >
                    <p
                      className="SubHeadingstyle___default-qgikex-0 cDJTT"
                    >
                      About Â£0
                    </p>
                  </SubHeadingstyle___default>
                </_default>
                <GradientCardstyle___default>
                  <div
                    className="GradientCardstyle___default-sc-17ewli4-0 dAlLHb"
                  >
                    <span>
                      Back it because you believe in it.
                    </span>
                    <p>
                      Support the project for no rewards, just because it speaks to you.
                    </p>
                  </div>
                </GradientCardstyle___default>
                <ContinueButton>
                  <ContinueButtonstyle___default>
                    <button
                      className="ContinueButtonstyle___default-mwgchk-0 hTA-dXx"
                    >
                      Continue
                    </button>
                  </ContinueButtonstyle___default>
                </ContinueButton>
              </Component>
            </WithTheme(Component)>
          </div>
        </Containerstyle___default>
      </_default>
    </GradientCard>
  </ThemeProvider>
</Theme>
`;
